usl_pipeline/cloud_functions/main.py

```diff
@@ -184,25 +184,31 @@ def write_flood_scenario_metadata_and_features(
         metastore.FloodScenarioConfig(
             gcs_path=f"gs://{config_blob.bucket.name}/{config_blob.name}",
             as_vector_gcs_path=f"gs://{vector_blob.bucket.name}/{vector_blob.name}",
+            num_rainfall_entries=length,
             # File names should be in the form <parent_config_name>/<file_name>
             parent_config_name=file_name.parent.name,
         ).set(db)
 
 
-def _rain_config_as_vector(rain_fd: TextIO) -> NDArray:
+def _rain_config_as_vector(rain_fd: TextIO) -> Tuple[NDArray, int]:
     """Converts the rainfall config contents into a vector describing the rainfall.
 
     Args:
       rain_fd: The file containing the CityCAT rainfall configuration.
 
     Returns:
-      The rainfall pattern as a numpy array.
+      A tuple of (rainfall, length) where `rainfall` contains
+      the rainfall pattern as a numpy array, padded to an agreed-upon
+      length, and `length` represent the un-padded number entries in
+      the array (i.e. the length the array would be if it was not
+      padded.)
+
     """
     entries = config_readers.read_rainfall_amounts(rain_fd)
     return numpy.pad(
         numpy.array(entries, dtype=numpy.float32),
         (0, _RAINFALL_VECTOR_LENGTH - len(entries)),
```

Comment:
Do we need any protection in case there are more than _RAINFALL_VECTOR_LENGTH entries in the config file?

Reply:
That's a good point! I can add a defense around this in the config uploader over here, which is also how we ensure things like entries being five minutes apart:
https://github.com/UrbanSystemsLab/climateiq-flood-model/blob/75797dcf9dd83860e8930d16ccc17c9556eae13c/config_uploader/config_uploader/main.py#L130

Reply:
Added a check in here.